src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-join-auth-list/smb-join-auth-list.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, Input, OnInit, ViewChild } from '@angular/core';
+import { ActionLabelsI18n } from '~/app/shared/constants/app.constants';
+import { TableComponent } from '~/app/shared/datatable/table/table.component';
+import { CdTableAction } from '~/app/shared/models/cd-table-action';
+import { CdTableColumn } from '~/app/shared/models/cd-table-column';
+import { CdTableFetchDataContext } from '~/app/shared/models/cd-table-fetch-data-context';
+import { Permission } from '~/app/shared/models/permissions';
+import { AuthStorageService } from '~/app/shared/services/auth-storage.service';
+import { SMBJoinAuth } from '../smb.model';
+import { ListWithDetails } from '~/app/shared/classes/list-with-details.class';
+
+@Component({
+  selector: 'cd-smb-join-auth-list',
+  templateUrl: './smb-join-auth-list.component.html',
+  styleUrls: ['./smb-join-auth-list.component.scss']
+})
+export class SmbJoinAuthListComponent extends ListWithDetails implements OnInit {
+  @Input()
+  joinAuths: SMBJoinAuth[]
+  @ViewChild('table', { static: true })
+  table: TableComponent;
+  columns: CdTableColumn[];
+  permission: Permission;
+  tableActions: CdTableAction[];
+  context: CdTableFetchDataContext;
+
+  // joinAuth$: Observable<SMBJoinAuth[]>;
+  // subject$ = new BehaviorSubject<SMBJoinAuth[]>([]);
+
+  constructor(
+    private authStorageService: AuthStorageService,
+    public actionLabels: ActionLabelsI18n
+  ) {
+    super();
+    this.permission = this.authStorageService.getPermissions().smb;
+  }
+
+  ngOnInit() {
+    this.columns = [
+      {
+        name: $localize`ID`,
+        prop: 'auth_id',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Username`,
+        prop: 'auth.username',
+        flexGrow: 2
+      },
+      {
+        name: $localize`Password`,
+        prop: 'auth.password',
+        flexGrow: 2
+      },
```

Comment:
we  can add copy to clipboard here for ease of access to user

Reply:
removed passwords as for now due to feedback gotten last demo, might add them later hidden by default with an option to see them